src/scatter_buildblock/ScatterEstimationByBin.cxx

```diff
@@ -11,447 +11,1274 @@
+        if ( this->do_average_at_2)
+        {
+            if (i_scat_iter == 2) // do average 0 and 1
+            {
+                if (is_null_ptr(act_image_for_averaging))
+                    error("Storing the first actibity estimate has failed at some point.");
+
+                //                if (this->run_debug_mode)// Write the previous estimate
+                //                {
+                //                    std::string output1= "./extras/act_image_for_averaging";
+                //                    OutputFileFormat<DiscretisedDensity<3,float> >::default_sptr()->
+                //                            write_to_file(output1, *act_image_for_averaging);
+                //                }
+                std::transform(act_image_for_averaging->begin_all(), act_image_for_averaging->end_all(),
+                               this->activity_image_lowres_sptr->begin_all(),
+                               this->activity_image_lowres_sptr->begin_all(),
+                               std::plus<float>());
+
+                //                if(this->run_debug_mode)// write after sum
+                //                {
+                //                    std::string output1= "./extras/summed_for_averaging";
+                //                    OutputFileFormat<DiscretisedDensity<3,float> >::default_sptr()->
+                //                            write_to_file(output1, *this->activity_image_lowres_sptr);
+                //                }
+
+                pow_times_add divide_by_two (0.0f, 0.5f, 1.0f, NumericInfo<float>().min_value(),
+                                             NumericInfo<float>().max_value());
+
+                in_place_apply_function(*this->activity_image_lowres_sptr, divide_by_two);
+
+                //                if(this->run_debug_mode)// write after division.
+                //                {
+                //                    std::string output1= "./extras/recon_0_1";
+                //                    OutputFileFormat<DiscretisedDensity<3,float> >::default_sptr()->
+                //                            write_to_file(output1, *this->activity_image_lowres_sptr);
+                //                }
+
+                // Have averaged ... turn false
+                this->do_average_at_2 = false;
+            }
+        }
+
+        //TODO: filter ... if FBP
+
+        info("Scatter simulation in progress...");
+        //        this->scatter_simulation_sptr->set_activity_image_sptr(this->activity_image_lowres_sptr);
+        this->scatter_simulation_sptr->process_data();
+        //        this->scatter_simulation_sptr->set_output_proj_data(unscaled);
+
+        if(this->run_debug_mode) // Write unscaled scatter sinogram
+        {
+            std::stringstream convert;   // stream used for the conversion
+            int lab = i_scat_iter+1;
+            convert << "./extras/unscaled_" << lab;
```

Comment:
fails on Windows. also fails if "extras" directory doesn't exist yet. make the code simple and just file in local directory (or better, make a variable with the output_prefix)